test/powershell/Language/Parser/TypeAccelerator.Tests.ps1

```diff
@@ -1,25 +1,437 @@
+                @{
+                    Accelerator = 'powershell'
+                    Type        = [System.Management.Automation.Powershell]
+                }
+                @{
+                    Accelerator = 'runspacefactory'
+                    Type        = [System.Management.Automation.Runspaces.RunspaceFactory]
+                }
+                @{
+                    Accelerator = 'runspace'
+                    Type        = [System.Management.Automation.Runspaces.Runspace]
+                }
+                @{
+                    Accelerator = 'initialsessionstate'
+                    Type        = [System.Management.Automation.Runspaces.InitialSessionState]
+                }
+                @{
+                    Accelerator = 'psscriptmethod'
+                    Type        = [System.Management.Automation.PSScriptMethod]
+                }  
+                @{
+                    Accelerator = 'psscriptproperty'
+                    Type        = [System.Management.Automation.PSScriptProperty]
+                }
+                @{
+                    Accelerator = 'psnoteproperty'
+                    Type        = [System.Management.Automation.PSNoteProperty]
+                }
+                @{
+                    Accelerator = 'psaliasproperty'
+                    Type        = [System.Management.Automation.PSAliasProperty]
+                }
+                @{
+                    Accelerator = 'psvariableproperty'
+                    Type        = [System.Management.Automation.PSVariableProperty]
+                }
+            )
+        
+            if ( $IsCoreCLR )
+            {
+                $totalAccelerators = 89 
+            } else
+            {
+                $totalAccelerators = 94
 
-    It "Basic type accelerator usage" {
-        [msft_2174855] | Should Be ([int])
-    }
+                $nonCoreTypeAcceleratorTestCases = @(
+                    @{
+                        Accelerator = 'adsi'
+                        Type        = [System.DirectoryServices.DirectoryEntry]
+                    }
+                    @{
+                        Accelerator = 'adsisearcher'
+                        Type        = [System.DirectoryServices.DirectorySearcher]
+                    }
+                    @{
+                        Accelerator = 'wmiclass'
+                        Type        = [System.Management.ManagementClass]
+                    }
+                    @{
+                        Accelerator = 'wmi'
+                        Type        = [System.Management.ManagementObject]
+                    }
+                    @{
+                        Accelerator = 'wmisearcher'
+                        Type        = [System.Management.ManagementObjectSearcher]
+                    }
+                )
+            }
+        }
+
+        It 'Should have all the type accelerators' {
+            $TypeAccelerators.Count | Should be $totalAccelerators
+        }

```

Comment:
Thanks for clarify!

>There are tests to check each of the accelerators, so we will see a failure if one is replaced with another. This "count" test makes sure no new accelerator is added accidentally.

If we have explicit full accelerator list in the file we catch the problem - in  DefaultCommands.Tests.ps1  we compare two arrays for this.